libsolidity/NameAndTypeResolver.cpp

```diff
@@ -82,6 +82,7 @@ void NameAndTypeResolver::checkTypeRequirements(ContractDefinition& _contract)
 	for (ASTPointer<StructDefinition> const& structDef: _contract.getDefinedStructs())
 		structDef->checkValidityOfMembers();
 	_contract.checkTypeRequirements();
+	checkDuplicateFunctionDeclaration(_contract); // TODO:
```

Comment:
Thank you. I'll move it into `checkTypeRequirements()`. The reason of another round check is: if the overloaded function is not used, there's nowhere to check for duplication, so it has to be called somewhere after the type requirements are checked.